crates/context_aware_config/src/api/config/handlers.rs

```diff
@@ -536,8 +552,109 @@ async fn reduce_config(
     Ok(HttpResponse::Ok().json(config))
 }
 
+#[cfg(feature = "high-performance-mode")]
+#[get("/fast")]
+async fn get_config_fast(
+    tenant: Tenant,
+    state: Data<AppState>,
+) -> superposition::Result<HttpResponse> {
+    use fred::interfaces::MetricsInterface;
+
+    log::debug!("Started redis fetch");
+    let config_key = format!("{}::cac_config", *tenant);
+    let max_created_key = format!("{}::cac_config::max_created_at", *tenant);
+    let audit_id_key = format!("{}::cac_config::audit_id", *tenant);
+    let config_version_key = format!("{}::cac_config::config_version", *tenant);
+    let client = state.redis.next_connected();
+    let config = client.get::<String, String>(config_key.clone()).await;
+    let metrics = client.take_latency_metrics();
+    let network_metrics = client.take_network_latency_metrics();
+    log::trace!(
+        "Network metrics for config fetch in milliseconds :: max: {}, min: {}, avg: {}",
+        network_metrics.max,
+        network_metrics.min,
+        network_metrics.avg
+    );
+    log::trace!(
+        "Latency metrics for config fetch in milliseconds :: max: {}, min: {}, avg: {}",
+        metrics.max,
+        metrics.min,
+        metrics.avg
+    );
+    match config {
+        Ok(config) => {
+            let mut response = HttpResponse::Ok();
+            if let Ok(max_created_at) =
+                client.get::<String, String>(max_created_key).await
```

Comment:
Can we parallelize the redis get operations for max_created_at , audit_id and config_version after fetching the config , that would make it even faster , we can use tokio_join macro here : 
```
          let ( max_created_at, audit_id, config_version) = tokio::join!(
                state.redis.get::<String, String>(max_created_key.clone()),
                state.redis.get::<String, String>(audit_id_key.clone()),
                state.redis.get::<Option<i64>, String>(config_version_key.clone()),
            );
```

Reply:
Its fast enough that the overhead of parallelization would hurt it more than help

Reply:
Can we get a benchmark ?